### ISSUE TITLE
Tray Icon in Linux  Automatically  Disappears

### ISSUE THREAD

**Issue by Aswathips** [AvaloniaUI/Avalonia#13130]
tray icon in linux system automatically Disappears , starting of the program there is no issue after some time of work it will happen , i tested this issue in various ubuntu versions and there is no error log ( in apps exception log and linux syslog)

- OS:  Linux (Ubuntu 18 , 20)
- Avalonia-Version: [11.0.4 and below ]

**Comment by timunie** [comment]
can you test a minimal sample? If it still happens, please upload your sample here for some of us to double-check.

**Comment by Aswathips** [comment]
ok i will give you a sample repo

**Comment by Aswathips** [comment]
Here is the minimal sample. In this sample also the tray icon disapears.

https://github.com/Aswathips/TrayIcon_Test.git

**Comment by timunie** [comment]
Okay I tried your sample on my windows machine. I guess I know what happens here. If the user closes the last window, the App thinks it should shut down. However, in your case it is not what you want. So try to add this line and check if it helps in your situation:

```diff
    public override void OnFrameworkInitializationCompleted()
    {
        if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
        {
+           desktop.ShutdownMode = ShutdownMode.OnExplicitShutdown;
            
            _contex = DataContext as ApplicationViewModel;
            //other code
        }

        base.OnFrameworkInitializationCompleted();
    }
```

**Comment by Aswathips** [comment]
I already added it in program.cs file

  public static void Main(string[] args)
  {
      BuildAvaloniaApp()
          .StartWithClassicDesktopLifetime(args, ShutdownMode.OnExplicitShutdown);
  }

Actually my application doesn't shut down on window close. The application will run in the background even after the icon disapers from the system tray .

**Comment by timunie** [comment]
okay got it. We will investigate further.

**Comment by Aswathips** [comment]
Ok. Thank you

**Comment by Aswathips** [comment]
Hi, Is there is any update.

**Comment by timunie** [comment]
No, that issue has not high priority. So don't expect us to fix it in the next few days. If you need urgend support, we offer paid support.

**Comment by dsigninfinnium** [comment]
Any Updates on This ?

**Comment by timunie** [comment]
@dsigninfinnium see comment above

**Comment by Nayjest** [comment]
I had similar issue on Windows,
Thanks to @timunie 
`desktop.ShutdownMode = ShutdownMode.OnExplicitShutdown;` helped (my app have no main window)

**Comment by timunie** [comment]
> However, I suggest, it should be handled somehow by Avalonia devs, because such system behaviour doesn't feels good.

We cannot handle every scenario reliable in any way, so this is the way to go. I think it's a fair solution. Thanks for your understanding.

**Comment by janmechtel** [comment]
I also experience this problem on Linux. The icon disappears if I suspend the current session and then resume/login again.

Adding `.StartWithClassicDesktopLifetime(args, ShutdownMode.OnExplicitShutdown);` did not change anything about that behavior.

My app also runs in the background without any open windows.

**Comment by aw3-1337** [comment]
Any update on this?
Can confirm the issue is related to screen lock

**Comment by timunie** [comment]
@aw3-1337 if you have an idea where this is happening, a PR would be welcome, since you can reproduce.

**Comment by hiddenblue** [comment]
When I executd the avalonia program in ubuntu desktop with "sudo -E", the TrayIcon was missing.  this is reproducible. (why using sudo -E ? because my program need sudo or root privilege to manage other sub process. most GUi application cannot work with normal user environment variables, which is provided by -E parameter)
If you remove the elevating code, the trayIcon works.

my project link https://github.com/hiddenblue/Avalonia.CpuLimiter/tree/Dev

![image](https://github.com/user-attachments/assets/0f84f8c3-ac16-4767-9adf-1a59c19e5f0a)

**Comment by SweetLoverFT** [comment]
> 当我在 Ubuntu 桌面系统上使用“sudo -E”命令运行 Avalonia 程序时，托盘图标消失了。这个问题可以复现。（为什么要使用 sudo -E？因为我的程序需要 sudo 或 root 权限来管理其他子进程。大多数 GUI 应用程序无法使用普通用户环境变量，而这些环境变量是由 -E 参数提供的。） 如果移除提升权限的代码，托盘图标就能正常工作。
> 
> 我的项目链接[：https://github.com/hiddenblue/Avalonia.CpuLimiter/tree/Dev](https://github.com/hiddenblue/Avalonia.CpuLimiter/tree/Dev)
> 
> ![image](https://github.com/user-attachments/assets/0f84f8c3-ac16-4767-9adf-1a59c19e5f0a)

TrayIcon doesn't support run in super user mode, sudo or 4775 privilege will result in disappearance for TrayIcon
